File: string/485-lengthOfLongestSubstring.cpp
#include<string>
#include<set>
#include<iostream>

using SET=std::set<char>;
using STR=std::string;


int lengthOfLongestSubstring(const STR& s)
{
    int sz=s.size();
    int r=0;
    SET win{};
    int ans{};
    while(r<sz)
    {
        while(win.find(s[r])!=win.end())
        {
            auto it=win.begin();
            win.erase(it);
        }
        win.insert(s[r]);
        ans=std::max(ans,int(win.size()));
        r+=1;
    }
    return ans;
}

int main()
{
    STR s{"abcdebbc"};
    std::cout<<lengthOfLongestSubstring(s);
}

File: string/485-lengthOfLongestSubstring.py


def longestSubString(s):
    n=len(s)
    l=r=0
    ans=0
    win=set()
    while r<n:
        while s[r] in win:
            win.remove(s[l])
            l+=1
        win.add(s[r])
        ans=max(ans,len(win))
        r+=1
    return ans

s="abcbbc"
print(longestSubString(s))